tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -117,7 +118,9 @@ int32_t tiledb_filestore_schema_create(
     // create copies of the underlying core objects from within
     // the cpp objects constructed here
     (*array_schema)->array_schema_ = make_shared<tiledb::sm::ArraySchema>(
-        HERE(), tiledb::sm::ArrayType::DENSE);
+        HERE(),
+        make_shared<tiledb::sm::MemoryTracker>(HERE()),
```

Comment:
I'd probably stick with `ARRAY_CREATE` for the tracker you'll have to instantiate here.